File: BigInteger/results/BigInteger.cpp
#include <one.hpp>
#include <iostream>

class MathUtils {
  public:
    static int calc(int n) {
        int result = 1;
        for (int i = 2; i <= n; i++) {
            result = result * i;
            if (result > 10) {
                result = result >> 2;
            }
        }
        return result;
    }
    
    static sp<OneBigInteger> calcBig(int n) {
        auto result = OneBigInteger::fromInt(1);
        for (int i = 2; i <= n; i++) {
            result = result * i + 123;
            result = result + result;
            if (result > 10) {
                result = result >> 2;
            }
        }
        return result;
    }

  private:
};

int main()
{
    
    cout << (string() + "5 -> " + to_string(MathUtils::calc(5)) + ", 24 -> " + MathUtils::calcBig(24)) << endl;
    return 0;
}

File: BigInteger/results/BigInteger.swift
class MathUtils {
  class func calc(n: Int) -> Int {
      var result = 1
      var i = 2
      while i <= n {
          result = result * i
          if result > 10 {
              result = result >> 2
          }
          i += 1
      }
      return result
  }

  class func calcBig(n: Int) -> OneBigInteger? {
      var result: OneBigInteger? = OneBigInteger.fromInt(value: 1)
      var i = 2
      while i <= n {
          result = result * i + 123
          result = result + result
          if result > 10 {
              result = result >> 2
          }
          i += 1
      }
      return result
  }
}


print("5 -> \(MathUtils.calc(n: 5)), 24 -> \(MathUtils.calcBig(n: 24)!)")